core/src/ext-data.c

```diff
@@ -45,13 +50,47 @@ crsql_ExtData *crsql_newExtData(sqlite3 *db, unsigned char *siteIdBuffer) {
   pExtData->pragmaSchemaVersion = -1;
   pExtData->pragmaDataVersion = -1;
   pExtData->pragmaSchemaVersionForTableInfos = -1;
-  pExtData->siteId = siteIdBuffer;
   pExtData->pDbVersionStmt = 0;
   pExtData->tableInfos = 0;
   pExtData->rowsImpacted = 0;
   pExtData->updatedTableInfosThisTx = 0;
   crsql_init_table_info_vec(pExtData);
 
+  sqlite3_stmt *pStmt;
+
+  rc += sqlite3_prepare_v2(db,
+                           "SELECT ltrim(key, 'config.'), value FROM "
+                           "crsql_master WHERE key LIKE 'config.%';",
+                           -1, &pStmt, 0);
+
+  if (rc != SQLITE_OK) {
+    crsql_freeExtData(pExtData);
+    return 0;
+  }
+
+  // set defaults!
+  pExtData->mergeEqualValues = 0;
+
+  while (sqlite3_step(pStmt) == SQLITE_ROW) {
+    const unsigned char *name = sqlite3_column_text(pStmt, 0);
+    int colType = sqlite3_column_type(pStmt, 1);
+
+    if (strcmp("merge-equal-values", (char *)name) == 0) {
+      if (colType == SQLITE_INTEGER) {
+        const int value = sqlite3_column_int(pStmt, 1);
+        pExtData->mergeEqualValues = value;
+      } else {
+        // broken setting...
+        crsql_freeExtData(pExtData);
+        return 0;
+      }
+    } else {
+      // unhandled config setting
```

Comment:
return an error?

Reply:
I wasn't quite sure which way to go there. I didn't want to be too strict for the list of supported config settings, but I did want to be strict for each setting's value type.